src/app/ConcreteAttributePath.h

```diff
@@ -144,9 +144,15 @@ struct ConcreteDataAttributePath : public ConcreteAttributePath
 
     bool MatchesConcreteAttributePath(const ConcreteAttributePath & aOther) { return ConcreteAttributePath::operator==(aOther); }
 
-    bool operator==(const ConcreteDataAttributePath & aOther) const = delete;
-    bool operator!=(const ConcreteDataAttributePath & aOther) const = delete;
-    bool operator<(const ConcreteDataAttributePath & aOther) const  = delete;
+    bool operator==(const ConcreteDataAttributePath & aOther) const
+    {
+        return ConcreteAttributePath::operator==(aOther) && (mListIndex == aOther.mListIndex) && (mListOp == aOther.mListOp) &&
```

Comment:
documentation says that mList Index is only valid if mListOp is a list operation, otherwise it is to be ignored.

This seems to imply this logic should apply to equality operators too (or do we guarantee it is 0 if list operation is NotList)?

Reply:
This is guaranteed by our constructors (mListOp and mListIndex are only ever initialized together). But since they are public members there's nothing prevent some calling code from creating an invalid ConcreteDataAttributePath directly. If that's a concern I think it's worth refactoring this struct so that invalid states are unrepresentable.